### ISSUE TITLE
`json(path)` and `cbor(path)` do not support UTF8 with BOM

### ISSUE THREAD

**Issue by miyaji255** [typst/typst#3157]
### Description

`json(path)`, `cbor(path)` cannot read a file with utf8-bom.

### Operating system

Windows

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enter-tainer** [comment]
typst only supports utf-8. maybe it should be mentioned in the doc of `json`. for cbor, IIRC it is a binary format so I dont get the encoding here.

**Comment by Enter-tainer** [comment]
you may take a look at the doc of read and str. they both only support utf8.

**Comment by Enter-tainer** [comment]
yes, i suppose json(path) and json(str) share the same implementation so no wonder if they support the same encoding

**Comment by Myriad-Dreamin** [comment]
Typst does support loading files containing BOM heading with `#import` or `#include` (source files). This is because it believes that source files are text encoding so to safely and smartly remove the BOM. Similarly, we can also expand this behavior to other text files, including `json(path)`.

But I personally don't like to process BOM here and there. My suggestion is writing your own json function.

```typst
#let my-json-loader(file) = json.decode(remove-bom(read(file)))
```

**Comment by laurmaedje** [comment]
Looks like it's inconsistent. I'm not sure whether we should just never remove BOM or do it smartly for text files.
